matchzoo/datapack.py

```diff
@@ -0,0 +1,114 @@
+"""Matchzoo DataPack, pair-wise tuple (feature) and context as input."""
+
+import typing
+from pathlib import Path
+
+import dill
+import pandas as pd
+
+
+class DataPack(pd.DataFrame):
+    """
+    Matchzoo DataPack data structure, store dataframe and context.
+
+    Example:
+        >>> # features, context generate by processors.
+        >>> features = [([1,3], [2,3]), ([3,0], [1,6])]
+        >>> context = {'vocab_size': 2000}
+        >>> dp = DataPack(data=features,
+        ...               context=context)
+        >>> dp.context
+        {'vocab_size': 2000}
+        >>> # sample without replacement for generation.
+        >>> type(dp.sample(1))
+        <class 'matchzoo.datapack.DataPack'>
+        >>> dp.size
+        2
+        >>> features, context = dp.unpack()
+    """
+
+    _metadata = ['context']
+
+    DATA_FILENAME = 'data.pkl'
+    CONTEXT_FILENAME = 'context.pkl'
+
+    def __init__(self,
+                 data: list,
+                 context: dict={},
+                 index: list= None,
+                 columns: list=['text_left', 'text_right'],
+                 dtype: object=None,
+                 copy: bool=True):
+        """Initialize."""
+        super(DataPack, self).__init__(data=data,
+                                       index=index,
+                                       columns=columns,
+                                       dtype=dtype,
+                                       copy=copy)
+        if self.shape[1] != 2:
```

Comment:
This constrain means that DataPack can process two texts data, however if we have additional features of text pairs, such as TF-IDF, BM25 ... 

Maybe remove this constrain?

Reply:
@pl8787 got it.